nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -0,0 +1,1067 @@
+                                        if (queryTimeout > 0) {
+                                            throw se;
+                                        }
+                                    }
+
+                                    Record currentRecord;
+                                    while ((currentRecord = recordParser.nextRecord()) != null) {
+                                        Object sql = currentRecord.getValue(sqlField);
+                                        if (sql != null && !StringUtils.isEmpty((String) sql)) {
+                                            // Execute the statement as-is
+                                            s.execute((String) sql);
+                                        } else {
+                                            log.error("Record had no (or null) value for Field Containing SQL: {}, flowfile {} will be penalized and routed to failure",
+                                                    new Object[]{sqlField, flowFile});
+                                            flowFile = session.penalize(flowFile);
+                                            session.transfer(flowFile, REL_FAILURE);
+                                        }
+                                    }
+                                    session.transfer(flowFile, REL_SUCCESS);
+                                    session.getProvenanceReporter().send(flowFile, jdbcURL);
+                                } catch (final SQLException e) {
+                                    log.error("Unable to update database due to {}, flowfile {} will be penalized and routed to failure", new Object[]{e.getMessage(), flowFile}, e);
+                                    flowFile = session.penalize(flowFile);
+                                    session.transfer(flowFile, REL_FAILURE);
+                                }
+                            } else {
+                                log.warn("Record schema does not contain Field Containing SQL: {}, flowfile {} will be penalized and routed to failure", new Object[]{sqlField, flowFile});
+                                flowFile = session.penalize(flowFile);
+                                session.transfer(flowFile, REL_FAILURE);
+                            }
+                        }
+
+                    } else {
+                        // Ensure the table name has been set, the generated SQL statements (and TableSchema cache) will need it
+                        if(StringUtils.isEmpty(tableName)) {
+                            log.error("Cannot process {} because Table Name is null or empty; penalizing and routing to failure", new Object[]{flowFile});
+                            flowFile = session.penalize(flowFile);
+                            session.transfer(flowFile, REL_FAILURE);
+                            return;
+                        }
+
+                        final boolean includePrimaryKeys = UPDATE_TYPE.equals(statementType) && updateKeys == null;
```

Comment:
When I tried to update a table which has a primary key column, I got:
```
routing to failure:
org.apache.nifi.processor.exception.ProcessException: Table 'xxx' does not have a Primary Key and no Update Keys were specified
```

This `UPDATE_TYPE.equals(statementType)` should use equalsIgnoreCase as other code do.